source/common/config/utility.cc

```diff
@@ -2,9 +2,30 @@
 
 #include "common/protobuf/protobuf.h"
 
+#include "spdlog/spdlog.h"
+
 namespace Envoy {
 namespace Config {
 
+void Utility::checkClusterAndLocalInfo(const std::string& error_prefix,
+                                       const std::string& cluster_name,
+                                       Upstream::ClusterManager& cm,
+                                       const LocalInfo::LocalInfo& local_info) {
+  if (!cm.get(cluster_name)) {
+    throw EnvoyException(fmt::format("{}: unknown cluster '{}'", error_prefix, cluster_name));
+  }
+
+  checkLocalInfo(error_prefix, local_info);
+}
+
+void Utility::checkLocalInfo(const std::string& error_prefix,
```

Comment:
Out of curiosity, why can't this be done early on during initialization? Can local_info change dynamically?